src/impl/conv_direct.jl

```diff
@@ -81,6 +81,11 @@ function conv_direct!(
     # Use `calc_padding_regions` to determine where we do or don't need to worry about padding
     padded_regions, central_region = calc_padding_regions(cdims)
 
+    # Set outputs to zero to support custom datatypes (https://github.com/FluxML/NNlib.jl/issues/490)
+    if iszero(beta)
+        y = fill!(y, zero(yT))
```

Comment:
As [I mentioned in the original issue](https://github.com/FluxML/NNlib.jl/issues/490#issuecomment-1522434567), naive `fill!`/`fill` doesn't necessarily work with e.g. `BigInt`:

```
julia> a = Vector{BigInt}(undef, 2)
2-element Vector{BigInt}:
 #undef
 #undef

julia> fill!(a, zero(BigInt))
2-element Vector{BigInt}:
 0
 0

julia> a[1] ===a[2]
true
```

I don't know how important these kinds of usecases are for NNlib.jl though.

Reply:
Is that problematic here?
```Julia
julia> a = Vector{BigInt}(undef, 2)
2-element Vector{BigInt}:
 #undef
 #undef

julia> fill!(a, zero(BigInt))
2-element Vector{BigInt}:
 0
 0

julia> for (i, ai) in enumerate(a)
         a[i] = i + 0 * a[i]
       end

julia> a
2-element Vector{BigInt}:
 1
 2
```